src/main/java/io/fabric8/jenkins/openshiftsync/BuildDecisionHandler.java

```diff
@@ -103,21 +107,41 @@ public boolean shouldSchedule(Queue.Task p, List<Action> actions) {
           logger.info("Triggering Jenkins build on WorkflowJob " + wj.getFullName() + " due to running on kubernetes due to " + causeDescription(actions));
           return true;
         }
-        logger.info("Triggering OpenShift Build for WorkflowJob " + wj.getFullName() + " due to " + causeDescription(actions));
-        openShiftClient.buildConfigs()
-          .inNamespace(namespace).withName(jobName)
-          .instantiate(
-            new BuildRequestBuilder()
-              .withNewMetadata().withName(jobName).and()
-              .addNewTriggeredBy().withMessage("Triggered by Jenkins job at " + jobURL).and()
-              .build()
-          );
-        return false;
+
+        /* We are always finding the buildConfig by the the name of JenkinsJobName
+         This is failing in the case of repoName and buildConfigName not same and triggering a build by pushing to master
+         we need to find the buildConfig by gitLabel also as added in launcher backend and
+         we do in PipelineJobListener in case of initial creation
+         */
+        BuildConfig foundBuildConfig = OpenShiftUtils.getBuildConfig(getOpenShiftClient(), namespace, buildConfigname);
```

Comment:
could rename method e.g. `OpenShiftUtils.findBCbyNameOrLabel(...)` that explain what this function does in order to get rid of comments.
Let code say what it does than comments explaining it?